### PR TITLE
Protos

### PR DESCRIPTION
**Pull request by nivista**
This PR makes 3 changes:
* It adds a progress field to timers that also includes how many fires were skipped (for example after a long delay during rebalancing).
  * Current behavior to keep n or more semantics is that timers won't stop until completed reaches target executions, but skipped  field is still important.
* Add a messaging proto for kafka
  * This is stored in protos/internal and the output file is stored in internal/.gen. The output can't be in .gen/internal because then it can't be imported.
* Refactor.

### CODE REVIEW

File: protos/common.proto [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    string url = 1;
+    Method method = 2;
+    string body = 3;
+    map<string, string> headers = 4;
+}
+
+message IntervalConfig {
+    google.protobuf.Timestamp start_time = 1;
+    int32 interval = 2;
+    sint32 executions = 3;
```
Comment (andrewjdawson2016) [inline protos/common.proto]: what is sint32?

File: protos/common.proto [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+message CronConfig {
+    google.protobuf.Timestamp start_time = 1;
+    string cron = 2;
+    sint32 executions = 3;
```
Comment (andrewjdawson2016) [inline protos/common.proto]: I think you want some proto level constant that represents executions that fire forever - e.g. -1 means fire this cron schedule forever.

File: protos/internal/messaging.proto [hunk 3]
```diff
@@ -0,0 +1,12 @@
+syntax = "proto3";
+
+import "protos/common.proto";
+
+option go_package = "github.com/nivista/steady/internal/.gen/protos/messaging";
+
+message Timer {
```
Comment (andrewjdawson2016) [inline protos/internal/messaging.proto:7]: What if I want to delete a timer? What does that message look like?

**Review by andrewjdawson2016 — APPROVED** [review]
Looks good

File: protos/common.proto [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    oneof task {
+        HTTPConfig http_config = 1;
+    }
+}
+
+message Meta {
+    google.protobuf.Timestamp create_time = 1;
+}
+
+message Progress {
```
Comment (andrewjdawson2016) [inline protos/common.proto:24]: Is this progress struct specific to interval or will cron use the same struct?

File: protos/common.proto [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+message IntervalConfig {
+    google.protobuf.Timestamp start_time = 1;
+    int32 interval = 2;
+    sint32 executions = 3;
+}
+
+message CronConfig {
+    google.protobuf.Timestamp start_time = 1;
```
Comment (andrewjdawson2016) [inline protos/common.proto:53]: why do you need start_time for cron?

File: protos/common.proto [hunk 6]
```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
```
Comment (andrewjdawson2016) [inline protos/common.proto:1]: where is the fire forever constant?

**Review by andrewjdawson2016 — CHANGES_REQUESTED** [review]

File: protos/common.proto (hunk 5 above)
Comment (nivista) [reply protos/common.proto:53]: if someone wants to start a timer in the future

File: protos/common.proto (hunk 6 above)
Comment (nivista) [reply protos/common.proto:1]: i need to figure out how to do this

File: protos/common.proto (hunk 4 above)
Comment (nivista) [reply protos/common.proto:24]: this is for everyone.

File: protos/common.proto (hunk 2 above)
Comment (nivista) [reply protos/common.proto]: yes i need to figure out how to do that.

File: protos/internal/messaging.proto (hunk 3 above)
Comment (nivista) [reply protos/internal/messaging.proto:7]: that message is defined in services.proto

Comment (nivista) [reply protos/internal/messaging.proto:7]: oh ik what you mean in Kafka
that is just null.

Comment (nivista) [reply protos/internal/messaging.proto:7]: If i put a proto in for that message, log compaction wouldnt remove the timer

File: protos/common.proto (hunk 1 above)
Comment (nivista) [reply protos/common.proto]: signed int, so i can have -1 for executions

Comment (andrewjdawson2016) [reply protos/common.proto]: Ah makes total sense

File: protos/common.proto (hunk 2 above)
Comment (andrewjdawson2016) [reply protos/common.proto]: 👍

File: protos/internal/messaging.proto (hunk 3 above)
Comment (andrewjdawson2016) [reply protos/internal/messaging.proto:7]: Gotcha

File: protos/common.proto (hunk 4 above)
Comment (andrewjdawson2016) [reply protos/common.proto:24]: Okay great

File: protos/common.proto (hunk 5 above)
Comment (andrewjdawson2016) [reply protos/common.proto:53]: Gotcha

**Review by andrewjdawson2016 — APPROVED** [review]

**Pull request merged**